frontend/src/metabase/query_builder/selectors.js

```diff
@@ -162,23 +162,38 @@ const getNextRunParameterValues = createSelector(
       .filter(p => p !== undefined),
 );
 
+function normalizeClause(clause) {
+  return typeof clause.raw === "function" ? clause.raw() : clause;
+}
+
 // Certain differences in a query should be ignored. `normalizeQuery`
-// standardizes the query before comparision in `getIsResultDirty`.
-function normalizeQuery(query, tableMetadata) {
+// standardizes the query before comparison in `getIsResultDirty`.
+export function normalizeQuery(query, tableMetadata) {
   if (!query) {
     return query;
   }
-  if (query.query && tableMetadata) {
-    query = updateIn(query, ["query", "fields"], fields => {
-      fields = fields
-        ? // if the query has fields, copy them before sorting
-          [...fields]
-        : // if the fields aren't set, we get them from the table metadata
-          tableMetadata.fields.map(({ id }) => ["field", id, null]);
-      return fields.sort((a, b) =>
-        JSON.stringify(b).localeCompare(JSON.stringify(a)),
-      );
-    });
+  if (query.query) {
```

Comment:
This might be an opportunity to break this function down and reduce its cyclomatic complexity.

Reply:
100%, I'd be happy to open a followup PR after 41 is released to drop some risks